### PR TITLE
meson: 0.57.1 -> 0.59.2, gst-plugins-base 1.18.4 -> 1.18.5

### PR DESCRIPTION
**Pull request by Princemachiavelli**
https://mesonbuild.com/Release-notes-for-0-59-0.html

Updated meson for compatibility with newer packages such as wlroots 0.16
Updated gst-plugins-base for compatibility with meson >= 0.58

I just stumbled upon the meson version issue when building gamescope since it requires wlroots 0.16. https://github.com/Plagman/gamescope

nixpkgs-review is in progress although it might end up being more than practical to run on a single machine. i.e it seems this will trigger a rebuild of >40,000 packages just looking at nixpkgs-review now. :/

I have built meson itself & gst-plugins-base without issue.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Princemachiavelli** [comment]
I'm going to at least build systemd, sssd, and thunderbird as a sanity check. If anyone has any other packages that should be tested prior to merging let me know.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 140545` at 46e2480d run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>248 packages marked as broken and skipped:</summary>
  <ul>
    <li>OVMF-CSM</li>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>airtame</li>
    <li>ams-lv2</li>
    <li>apfelgrid</li>
    <li>applgrid</li>
    <li>artyFX</li>
    <li>asymptote</li>
    <li>autotrace</li>
    <li>beignet</li>
    <li>betaflight-configurator</li>
    <li>blastem</li>
    <li>bless</li>
    <li>bootiso</li>
    <li>bud</li>
    <li>cargo-rr</li>
    <li>cassandra_2_1</li>
    <li>cassandra_2_2</li>
    <li>chez</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>colmapWithCuda</li>
    <li>cura_stable</li>
    <li>deeptools</li>
    <li>devede</li>
    <li>displaycal</li>
    <li>dvd-slideshow</li>
    <li>easyeffects</li>
    <li>echidna</li>
    <li>electron_3</li>
    <li>electron_4</li>
    <li>electron_5</li>
    <li>embree</li>
    <li>embree2</li>
    <li>emuflight-configurator</li>
    <li>eterm</li>
    <li>evilvte</li>
    <li>exodus</li>
    <li>facter</li>
    <li>ffmpeg_2</li>
    <li>ffmpeg_2_8</li>
    <li>ffmpeg_3</li>
    <li>ffmpeg_3_4</li>
    <li>forge</li>
    <li>gb-backup</li>
    <li>git-dit</li>
    <li>gitin</li>
    <li>globalplatform</li>
    <li>globulation2</li>
    <li>gmtk</li>
    <li>gnome_mplayer</li>
    <li>gogoclient</li>
    <li>gopro</li>
    <li>gppcscconnectionplugin</li>
    <li>gpshell</li>
    <li>gpu-burn</li>
    <li>grass</li>
    <li>gtkd</li>
    <li>gtmess</li>
    <li>hip</li>
    <li>hplipWithPlugin_3_16_11</li>
    <li>hplip_3_16_11</li>
    <li>idris2</li>
    <li>impressive</li>
    <li>intecture-cli</li>
    <li>intel-compute-runtime</li>
    <li>intel-media-driver</li>
    <li>invoice2data</li>
    <li>isabelle</li>
    <li>jboss</li>
    <li>jfbpdf</li>
    <li>jfbview</li>
    <li>k2pdfopt</li>
    <li>k3d</li>
    <li>keyfinder</li>
    <li>libav</li>
    <li>libav_0_8</li>
    <li>libav_11</li>
    <li>libav_12</li>
    <li>libgroove</li>
    <li>librealsenseWithCuda</li>
    <li>libvirt_5_9_0</li>
    <li>libvmi</li>
    <li>linuxKernel.packages.hardkernel_4_14.evdi</li>
    <li>linuxKernel.packages.hardkernel_4_14.openrazer</li>
    <li>linuxKernel.packages.hardkernel_4_14.xpadneo</li>
    <li>linuxKernel.packages.hardkernel_4_14.zfs</li>
    <li>linuxKernel.packages.hardkernel_4_14.zfsStable</li>
    <li>linuxKernel.packages.hardkernel_4_14.zfsUnstable</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.evdi</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.oci-seccomp-bpf-hook</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.openrazer</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.xpadneo</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.zfs</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.zfsStable</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.zfsUnstable</li>
    <li>linuxPackages_hardkernel_latest.evdi</li>
    <li>linuxPackages_hardkernel_latest.openrazer</li>
    <li>linuxPackages_hardkernel_latest.xpadneo</li>
    <li>linuxPackages_hardkernel_latest.zfs</li>
    <li>linuxPackages_hardkernel_latest.zfsStable</li>
    <li>linuxPackages_hardkernel_latest.zfsUnstable</li>
    <li>linuxPackages_testing_bcachefs.evdi</li>
    <li>linuxPackages_testing_bcachefs.oci-seccomp-bpf-hook</li>
    <li>linuxPackages_testing_bcachefs.openrazer</li>
    <li>linuxPackages_testing_bcachefs.xpadneo</li>
    <li>linuxPackages_testing_bcachefs.zfs</li>
    <li>linuxPackages_testing_bcachefs.zfsStable</li>
    <li>linuxPackages_testing_bcachefs.zfsUnstable</li>
    <li>loxodo</li>
    <li>luxcorerender</li>
    <li>magma</li>
    <li>mame</li>
    <li>meo</li>
    <li>metamorphose2</li>
    <li>mrxvt</li>
    <li>mupdf_1_17</li>
    <li>mvapich</li>
    <li>nixui</li>
    <li>nvtop</li>
    <li>obs-studio-plugins.obs-ndi</li>
    <li>odpdown</li>
    <li>onedrive</li>
    <li>openems</li>
    <li>openrw</li>
    <li>openssh_gssapi</li>
    <li>openssh_hpn</li>
    <li>pcsxr</li>
    <li>pdfdiff</li>
    <li>pgadmin</li>
    <li>photoflow</li>
    <li>pig</li>
    <li>plover.stable</li>
    <li>poedit</li>
    <li>polymake</li>
    <li>processing</li>
    <li>pulp</li>
    <li>pulseeffects-legacy</li>
    <li>python38Packages.HTSeq</li>
    <li>python38Packages.TheanoWithCuda</li>
    <li>python38Packages.caffe</li>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.dask-gateway</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.dask-mpi</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.fipy</li>
    <li>python38Packages.foundationdb51</li>
    <li>python38Packages.foundationdb52</li>
    <li>python38Packages.foundationdb60</li>
    <li>python38Packages.foundationdb61</li>
    <li>python38Packages.foxdot</li>
    <li>python38Packages.ignite</li>
    <li>python38Packages.pycuda</li>
    <li>python38Packages.pygmt</li>
    <li>python38Packages.pymystem3</li>
    <li>python38Packages.pynvml</li>
    <li>python38Packages.pyrealsense2WithCuda</li>
    <li>python38Packages.python-csxcad</li>
    <li>python38Packages.python-openems</li>
    <li>python38Packages.pythonnet</li>
    <li>python38Packages.pytorchWithCuda</li>
    <li>python38Packages.spacy</li>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.stumpy</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python38Packages.textacy</li>
    <li>python38Packages.thinc</li>
    <li>python39Packages.HTSeq</li>
    <li>python39Packages.TheanoWithCuda</li>
    <li>python39Packages.caffe</li>
    <li>python39Packages.cnvkit</li>
    <li>python39Packages.dask-gateway</li>
    <li>python39Packages.dask-jobqueue</li>
    <li>python39Packages.dask-ml</li>
    <li>python39Packages.dask-mpi</li>
    <li>python39Packages.dask-xgboost</li>
    <li>python39Packages.fipy</li>
    <li>python39Packages.foundationdb51</li>
    <li>python39Packages.foundationdb52</li>
    <li>python39Packages.foundationdb60</li>
    <li>python39Packages.foundationdb61</li>
    <li>python39Packages.foxdot</li>
    <li>python39Packages.ignite</li>
    <li>python39Packages.pycuda</li>
    <li>python39Packages.pygmt</li>
    <li>python39Packages.pymystem3</li>
    <li>python39Packages.pynvml</li>
    <li>python39Packages.pyrealsense2WithCuda</li>
    <li>python39Packages.python-csxcad</li>
    <li>python39Packages.python-openems</li>
    <li>python39Packages.pythonnet</li>
    <li>python39Packages.pytorchWithCuda</li>
    <li>python39Packages.spacy</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.streamz</li>
    <li>python39Packages.stumpy</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>python39Packages.textacy</li>
    <li>python39Packages.thinc</li>
    <li>qemu_xen</li>
    <li>qemu_xen-light</li>
    <li>qemu_xen_4_10</li>
    <li>qemu_xen_4_10-light</li>
    <li>qlandkartegt</li>
    <li>qubes-core-vchan-xen</li>
    <li>ring-daemon</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>root5</li>
    <li>rxvt</li>
    <li>scribus</li>
    <li>secretscanner</li>
    <li>shellinabox</li>
    <li>shotgun</li>
    <li>siril</li>
    <li>sonic-pi</li>
    <li>spark</li>
    <li>spark2</li>
    <li>spark3</li>
    <li>split2flac</li>
    <li>spotifywm</li>
    <li>spring</li>
    <li>springLobby</li>
    <li>stepmania</li>
    <li>tilix</li>
    <li>tlaps</li>
    <li>torchat</li>
    <li>truecrack-cuda</li>
    <li>truvari</li>
    <li>twinkle</li>
    <li>unetbootin</li>
    <li>unicap</li>
    <li>vbam</li>
    <li>vcs</li>
    <li>wimlib</li>
    <li>winpdb</li>
    <li>woeusb</li>
    <li>xfitter</li>
    <li>xorg.libAppleWM</li>
    <li>xpdf</li>
    <li>yosys-bluespec</li>
    <li>zandronum</li>
  </ul>
</details>
<details>
  <summary>20053 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>CHOWTapeModel</li>
    <li>DisnixWebService</li>
    <li>EBTKS</li>
    <li>EmptyEpsilon</li>
    <li>Fabric (python39Packages.Fabric)</li>
    <li>MIDIVisualizer</li>
    <li>MMA</li>
    <li>OVMF</li>
    <li>OVMF-secureBoot</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>58 packages built successfully:</summary>
  <ul>
    <li>cargo</li>
    <li>cryptsetup</li>
    <li>epoxy</li>
    <li>fribidi</li>
    <li>gi-docgen</li>
    <li>gitMinimal</li>
    <li>gtk-doc (gnome2.gtkdoc)</li>
    <li>libdrm</li>
    <li>libfido2</li>
    <li>libglvnd</li>
    <li>libpsl</li>
    <li>libsysprof-capture</li>
    <li>libusb1</li>
    <li>libvdpau</li>
    <li>lvm2</li>
    <li>meson</li>
    <li>openssh</li>
    <li>python38Packages.meson</li>
    <li>python39Packages.cryptography</li>
    <li>python39Packages.meson</li>
    <li>python39Packages.pyftpdlib</li>
    <li>python39Packages.pyopenssl</li>
    <li>python39Packages.pyroma</li>
    <li>python39Packages.requests</li>
    <li>python39Packages.responses</li>
    <li>python39Packages.service-identity</li>
    <li>sphinx (python39Packages.sphinx)</li>
    <li>python39Packages.trustme</li>
    <li>rlottie</li>
    <li>systemdMinimal</li>
    <li>xkeyboard_config (xorg.xkeyboardconfig)</li>
    <li>xorg.libICE</li>
    <li>xorg.libSM</li>
    <li>xorg.libX11</li>
    <li>xorg.libXScrnSaver</li>
    <li>xorg.libXau</li>
    <li>xorg.libXcomposite</li>
    <li>xorg.libXcursor</li>
    <li>xorg.libXdamage</li>
    <li>xorg.libXdmcp</li>
    <li>xorg.libXext</li>
    <li>xorg.libXfixes</li>
    <li>xorg.libXi</li>
    <li>xorg.libXinerama</li>
    <li>xorg.libXmu</li>
    <li>xorg.libXrandr</li>
    <li>xorg.libXrender</li>
    <li>xorg.libXt</li>
    <li>xorg.libXtst</li>
    <li>xorg.libXv</li>
    <li>xorg.libXxf86vm</li>
    <li>xorg.libxcb</li>
    <li>xorg.xcbutil</li>
    <li>xorg.xcbutilimage</li>
    <li>xorg.xcbutilkeysyms</li>
    <li>xorg.xcbutilrenderutil</li>
    <li>xorg.xcbutilwm</li>
    <li>xorg.xorgproto</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `writeTextDir`.
Near pkgs/development/tools/build-managers/meson/default.nix:3:3:
```
  |
3 | , writeTextDir
  |   ^
```
</li>
  </ul>
</details>

**Comment by jtojnar** [comment]
Due to this causing a mass rebuild, it needs to target staging.

**Comment by jtojnar** [comment]
I usually build fwupd and gnome-shell, or NixOS VM with GNOME if I can.

**Comment by Princemachiavelli** [comment]
I've updated the PR to target staging. I'll rebuild the packages: fwupd, gnome-shell, systemd, sssd, and thunderbird as time allows.

**Comment by Princemachiavelli** [comment]
* fwupd: builds OK
* sssd: fails but its also failing in hydra staging... I have a in-progress update to switch it to the 2.x branch.
* systemd: builds OK
* gnome-shell: fails with meson error but the error is : ERROR: Program 'dot' not found...
* thunderbird: TODO

Checking now if gnome-shell fails in clean staging nixpkgs.

**Comment by jtojnar** [comment]
Is that in gnome-shell derivation? I do not see any reference to dot in the meson files.

**Comment by Princemachiavelli** [comment]
The missing 'dot' error was in the libnice package which now needs graphviz. It looks like the Archlinux team had the same issue when meson was updated so it probably has something to do with documentation generation.

https://github.com/archlinux/svntogit-packages/commit/828b95566501b9918d85937a41da85709f506004#diff-37538beb61ff63edebbf735dfcf39e5d732f49183d6beb097169d971875ca422

**Comment by jtojnar** [comment]
Yeah, that is another instance of https://github.com/mesonbuild/meson/issues/8641. Adding graphviz is probably okay.

**Comment by Princemachiavelli** [comment]
I am also getting some failed tests in some python packages:

In the most recent case: python3.9-sh

```
======================================================================
ERROR: test_general_signal (test.FunctionalTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/sh-1.14.2/test.py", line 1546, in test_general_signal
    p = python(py.name, _out=agg, _tee=True)
  File "/build/sh-1.14.2/sh.py", line 1566, in __call__
    return RunningCommand(cmd, call_args, stdin, stdout, stderr)
  File "/build/sh-1.14.2/sh.py", line 822, in __init__
    self.wait()
  File "/build/sh-1.14.2/sh.py", line 879, in wait
    self.handle_command_exit_code(exit_code)
  File "/build/sh-1.14.2/sh.py", line 905, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1:

  RAN: /nix/store/g5rqxbchc59xrb3b5h41n8wh73l1sklk-python3-3.9.6/bin/python3.9 /build/tmp6xr_dtu1

  STDOUT:
0
1
2
3


  STDERR:
Traceback (most recent call last):
  File "/build/tmp6xr_dtu1", line 14, in <module>
    print(i)
  File "/build/tmp6xr_dtu1", line 8, in sig_handler
    print(10)
RuntimeError: reentrant call inside <_io.BufferedWriter name='<stdout>'>


----------------------------------------------------------------------
Ran 171 tests in 49.533s

FAILED (errors=1, skipped=3)
Test failed: <unittest.runner.TextTestResult run=171 errors=1 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=171 errors=1 failures=0>
```

Let me know if there's a better way to get a concise output of failing tests instead of nixpkgs-review's output.

**Comment by jtojnar** [comment]
I do not use nix-review but would expect `nix log -f . attributeThatFailed` to work.

**Comment by Princemachiavelli** [comment]
Besides the failing tests for a few python3 packages, gnome-shell builds and launches OK. The libnice fix is also in place.

I went ahead and updated the remaining gstreamer (gst_all_1.*) packages to 1.18.5 and they all built fine. It seemed like a good idea to update those packages together.

I am now going to go back and try to rerun tests for the packages that had test failures & then check if they are failing in regular staging.

**Comment by jmesmon** [comment]
FYI, [meson 0.60.1](https://github.com/mesonbuild/meson/releases/tag/0.60.1) has been released.

File: pkgs/development/libraries/gstreamer/bad/default.nix [hunk 1]
```diff
@@ -93,27 +93,18 @@
...
-    sha256 = "06ildd4rl6cynirv3p00d2ddf5is9svj4i7mkahldzhq24pq5mca";
+    sha256 = "sha256-oWSSO5Tw0IV4pvyuqsbgwF2niKRpA6EIaHDpykWtZ44=";
   };
 
   patches = [
     # Use pkgconfig to inject the includedirs
     ./fix_pkgconfig_includedir.patch
-  ] ++ lib.optionals stdenv.isDarwin [
-    # Fix “error: cannot initialize a parameter of type 'unsigned long *' with an rvalue of type 'typename std::remove_reference<decltype(*(&opencv_dilate_erode_type))>::type *' (aka 'volatile unsigned long *')” on Darwin.
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gstreamer/bad/default.nix:110]: Please remove the unused fetchpatch from inputs.

**Comment by Princemachiavelli** [comment]
Looks like the meson patches need to be updated to work with meson 0.60.1. I should be able to have those done today. I'll create a new PR for meson 0.60.1 and close this one.

I am/was planning to wait until after the 20.11 release (or at least Nov 12 when staging is open again) before requesting another review since we are now already in ZHF. Since the only broken package to my knowledge without this update is the latest wlroots I doubt there's any reason to force this into 20.11.

**Comment by SuperSandro2000** [comment]
> I'll create a new PR for meson 0.60.1 and close this one.

Why can't we update this one?

**Comment by Princemachiavelli** [comment]
> > I'll create a new PR for meson 0.60.1 and close this one.
> 
> Why can't we update this one?

I named the source branch meson-0.59.2 and I don't think there is a way to switch or rename the source branch. I guess it doesn't make a practical difference, I could just update this branch and update the title.

**Comment by jtojnar** [comment]
Closing as superseded by https://github.com/NixOS/nixpkgs/pull/144779.

**Pull request closed without merging**